packages/create-emotion/types/index.d.ts

```diff
@@ -1,107 +1,53 @@
-  | undefined | null | boolean | string
-  | { [key: string]: undefined | null | boolean | string }
-  | ArrayClassNameArg
-  ;
-
-export interface StyleSheet {
-  inject(): void;
-  speedy(bool: boolean): void;
-  insert(rule: string, sourceMap?: string): void;
-  flush(): void;
-}
-
-export interface EmotionCache {
-  registered: {
-    [key: string]: string;
-  };
-  inserted: {
-    [key: string]: string;
-  };
-  nonce?: string;
-  key: string;
-}
-
-export interface Emotion {
-  flush(): void;
-  hydrate(ids: Array<string>): void;
-  cx(...classNames: Array<ClassNameArg>): string;
-  merge(className: string, sourceMap?: string): string;
-  getRegisteredStyles(registeredStyles: Array<string>, classNames: string): string;
-  css(...args: Array<Interpolation>): string;
-  injectGlobal(...args: Array<Interpolation>): void;
-  keyframes(...args: Array<Interpolation>): string;
-  sheet: StyleSheet;
-  caches: EmotionCache;
-}
-
-export interface EmotionBaseContext {
-  __SECRET_EMOTION__?: Emotion;
-}
+export {
+  ArrayInterpolation,
+  ComponentSelector,
+  FunctionInterpolation,
+  ObjectInterpolation
+} from '@emotion/serialize'
 
-export interface EmotionContext extends EmotionBaseContext {
-  [key: string]: any;
-}
+export { EmotionCache, Interpolation, Options, RegisteredCache, StyleSheet }
 
-export type StylisPlugins =
+export interface ArrayClassNamesArg extends Array<ClassNamesArg> {}
+export type ClassNamesArg =
+  | undefined
   | null
-  | ((...args: Array<any>) => any)
-  | Array<(...args: Array<any>) => any>
-  ;
-
-export interface EmotionOptions {
-  nonce?: string;
-  stylisPlugins?: StylisPlugins;
-  prefix?: boolean | ((key: string, value: string, context: 1 | 2 | 3) => boolean);
-  key?: string;
-  container?: HTMLElement;
-}
-
-export default function createEmotion(context: EmotionContext, options?: EmotionOptions): Emotion;
-
-declare module 'react' {
-  interface HTMLAttributes<T> {
-    css?: Interpolation;
-  }
-}
+  | string
+  | boolean
+  | { [className: string]: boolean }
+  | ArrayClassNamesArg
 
-// Preact support for css prop
-declare global {
-  namespace JSX {
-    interface HTMLAttributes {
-      css?: Interpolation;
-    }
-  }
-}
+export interface Emotion {
+  css(template: TemplateStringsArray, ...args: Array<Interpolation>): string
+  css(...args: Array<Interpolation>): string
+  cx(...classNames: Array<ClassNamesArg>): string
+  flush(): void
+  hydrate(ids: Array<string>): void
+  injectGlobal(
+    template: TemplateStringsArray,
+    ...args: Array<Interpolation>
+  ): void
+  injectGlobal(...args: Array<Interpolation>): void
+  keyframes(
+    template: TemplateStringsArray,
+    ...args: Array<Interpolation>
+  ): string
+  keyframes(...args: Array<Interpolation>): string
+  sheet: StyleSheet
+  caches: EmotionCache
+  merge(
+    registered: RegisteredCache,
+    css: (registeredStyles: Array<string>) => string,
+    className: string
+  ): string
+  getRegisteredStyles: typeof getRegisteredStyles
```

Comment:
The types of `merge` and `getRegisteredStyles` are wrong.

https://github.com/emotion-js/emotion/blob/master/packages/create-emotion/src/index.js#L132-L133

Reply:
Oh, I missed the parameters of bind. Thank you!